packages/libp2p/src/connection-manager/auto-dial.ts

```diff
@@ -1,285 +0,0 @@
-import { PeerMap, PeerSet } from '@libp2p/peer-collections'
-import { PeerQueue } from '@libp2p/utils/peer-queue'
-import { toString as uint8ArrayToString } from 'uint8arrays/to-string'
-import { AUTO_DIAL_CONCURRENCY, AUTO_DIAL_DISCOVERED_PEERS_DEBOUNCE, AUTO_DIAL_INTERVAL, AUTO_DIAL_MAX_QUEUE_LENGTH, AUTO_DIAL_PEER_RETRY_THRESHOLD, AUTO_DIAL_PRIORITY, LAST_DIAL_FAILURE_KEY, MIN_CONNECTIONS } from './constants.js'
-import type { Libp2pEvents, Logger, ComponentLogger, TypedEventTarget, PeerStore, Startable, Metrics } from '@libp2p/interface'
-import type { ConnectionManager } from '@libp2p/interface-internal'
-
-interface AutoDialInit {
-  minConnections?: number
-  maxQueueLength?: number
-  autoDialConcurrency?: number
-  autoDialPriority?: number
-  autoDialInterval?: number
-  autoDialPeerRetryThreshold?: number
-  autoDialDiscoveredPeersDebounce?: number
-}
-
-interface AutoDialComponents {
-  connectionManager: ConnectionManager
-  peerStore: PeerStore
-  events: TypedEventTarget<Libp2pEvents>
-  logger: ComponentLogger
-  metrics?: Metrics
-}
-
-const defaultOptions = {
-  minConnections: MIN_CONNECTIONS,
-  maxQueueLength: AUTO_DIAL_MAX_QUEUE_LENGTH,
-  autoDialConcurrency: AUTO_DIAL_CONCURRENCY,
-  autoDialPriority: AUTO_DIAL_PRIORITY,
-  autoDialInterval: AUTO_DIAL_INTERVAL,
-  autoDialPeerRetryThreshold: AUTO_DIAL_PEER_RETRY_THRESHOLD,
-  autoDialDiscoveredPeersDebounce: AUTO_DIAL_DISCOVERED_PEERS_DEBOUNCE
-}
-
-export class AutoDial implements Startable {
-  private readonly connectionManager: ConnectionManager
-  private readonly peerStore: PeerStore
-  private readonly queue: PeerQueue<void>
-  private readonly minConnections: number
-  private readonly autoDialPriority: number
-  private readonly autoDialIntervalMs: number
-  private readonly autoDialMaxQueueLength: number
-  private readonly autoDialPeerRetryThresholdMs: number
-  private readonly autoDialDiscoveredPeersDebounce: number
-  private autoDialInterval?: ReturnType<typeof setInterval>
-  private started: boolean
-  private running: boolean
-  private readonly log: Logger
-
-  /**
-   * Proactively tries to connect to known peers stored in the PeerStore.
-   * It will keep the number of connections below the upper limit and sort
-   * the peers to connect based on whether we know their keys and protocols.
```

Comment:
do we have this peer sorting functionality elsewhere? anything we need to keep in here?

Also kind of related to @maschad 's original ask about potentially keeping this for some users: There may be a need to have an auto-dialer for swarm-like functionality, where you want to stay connected to a specific set of peers? Can users accomplish that without auto-dial today?

Reply:
Yes, tag them with `KEEP_ALIVE`.